src/lakefs_spec/spec.py

```diff
@@ -542,7 +542,7 @@ def ls(
         # Retry the API call with appended slash if the current result
         # is just a single directory entry only (not its contents).
         # This is useful to allow `ls("repo/branch/dir")` calls without a trailing slash.
-        if len(info) == 1 and info[0]["type"] == "directory":
+        if len(info) == 1 and info[0]["type"] == "directory" and info[0]["name"] == path + "/":
```

Comment:
Unless I'm missing something, this should always be true, since `path` was requested in the original `reference.objects()` call (L520 above).

Reply:
Not always. Only when we request path without "/" at the end. If we have only one directory in the requested path, it would be false.